Add a default image prefiller implementation

larryliu0820 opened the pull request:
Summary: As titled. I need to create an interface `IModule` for `Module` class to override, to make it test-able.

Differential Revision: D80063769




cc @mergennachin @cccclai @helunwencser @jackzhxng

jackzhxng commented on extension/llm/runner/image_prefiller.cpp:
```diff
@@ -0,0 +1,92 @@
+ * LICENSE file in the root directory of this source tree.
+ */
+
+// Given a image tensor, prefill the KV cache of LLaVA.
```
Remove LLaVaA

larryliu0820 commented on extension/llm/runner/multimodal_prefiller.cpp:
```diff
@@ -0,0 +1,126 @@
+  }
+
+  auto outputs_res =
+      ET_UNWRAP(text_decoder_runner_->decode(encoder_output, start_pos));
```
`encoder_output` will be the `token_embedding` result

jackzhxng replied:
You have `text_encoder_outputs` right now in the text if branch

larryliu0820 replied:
Yeah updated.

larryliu0820 committed:
Add a default image prefiller implementation (#13310)

Summary:

As titled. I need to create an interface `IModule` for `Module` class to override, to make it test-able.

Reviewed By: jackzhxng

Differential Revision: D80063769

larryliu0820 committed:
Retry -ULEAK_SANITIZER

larryliu0820 committed:
Retry 2

larryliu0820 committed:
Retry 3

larryliu0820 committed:
Retry 4

larryliu0820 committed:
Retry 5

larryliu0820 committed:
Retry 6
